src/editor/gridstack/gridstackModel.ts

```diff
@@ -0,0 +1,362 @@
+        sessionContext
+      ).catch(reason => console.error(reason));
+    } else if (this._type === 'markdown') {
+      (this._cell as MarkdownCell).inputHidden = false;
+      (this._cell as MarkdownCell).rendered = true;
+    }
+
+    this.update();
+  } */
+
+  private _updateCells(): void {
+    this._checkCellsMetadata();
+    this.contentChanged.emit(void 0);
+  }
+
+  private _checkMetadata(): void {
+    let data = this._context.model.metadata.get('extensions') as Record<
+      string,
+      any
+    >;
+
+    if (!data) {
+      data = {
+        jupyter_dashboards: {
+          version: 1,
+          activeView: VIEW,
+          views: {
+            grid_default: this._info
+          }
+        }
+      };
+    } else if (!data.jupyter_dashboards) {
+      data['jupyter_dashboards'] = {
+        version: 1,
+        activeView: VIEW,
+        views: {
+          grid_default: this._info
+        }
+      };
+    } else if (
+      !data.jupyter_dashboards.views[VIEW] ||
+      !('name' in data.jupyter_dashboards.views[VIEW]) ||
+      !('type' in data.jupyter_dashboards.views[VIEW]) ||
+      !('cellMargin' in data.jupyter_dashboards.views[VIEW]) ||
+      !('cellHeight' in data.jupyter_dashboards.views[VIEW]) ||
+      !('numColumns' in data.jupyter_dashboards.views[VIEW])
+    ) {
+      data.jupyter_dashboards.views[VIEW] = this._info;
+    } else {
+      this._info = data.jupyter_dashboards?.views[VIEW] as DashboardView;
+    }
+
+    this._context.model.metadata.set('extensions', data);
+  }
+
+  private _checkCellsMetadata(): void {
+    for (let i = 0; i < this._context.model.cells?.length; i++) {
+      const cell = this._context.model.cells.get(i);
+      this._checkCellMetadata(cell);
+    }
+  }
+
+  private _checkCellMetadata(cell: ICellModel): void {
+    let data = cell.metadata.get('extensions') as Record<string, any>;
+
+    if (!data) {
+      data = {
+        jupyter_dashboards: {
+          activeView: VIEW,
+          views: {
+            grid_default: {
+              hidden: true,
+              row: null,
+              col: null,
+              width: 2,
+              height: 2
+            }
+          }
+        }
+      };
+      cell.metadata.set('extensions', data);
+    } else if (!data.jupyter_dashboards) {
+      data['jupyter_dashboards'] = {
+        activeView: VIEW,
+        views: {
+          grid_default: {
+            hidden: true,
+            row: null,
+            col: null,
+            width: 2,
+            height: 2
+          }
+        }
+      };
+      cell.metadata.set('extensions', data);
+    } else if (
+      !data.jupyter_dashboards.views[VIEW] ||
+      !('hidden' in data.jupyter_dashboards.views[VIEW]) ||
+      !('row' in data.jupyter_dashboards.views[VIEW]) ||
+      !('col' in data.jupyter_dashboards.views[VIEW]) ||
+      !('width' in data.jupyter_dashboards.views[VIEW]) ||
+      !('height' in data.jupyter_dashboards.views[VIEW])
```

Comment:
Maybe this could be handled by a schema validator.

Something that can be investigated in a separate change.